ThirdParty/CMakeLists.txt

```diff
@@ -47,6 +47,33 @@ endif()
 
 if(OGS_USE_PYTHON)
     add_subdirectory(pybind11)
+
+    function(check_python_version_compatibility)
+        if(NOT EXISTS ${VTK_DIR}/VTKConfig.cmake)
+            return()
+        endif()
+
+        include(${VTK_DIR}/VTKConfig.cmake)
+
+        if(NOT ${VTK_WRAP_PYTHON})
+            return()
+        endif()
+        if(NOT EXISTS ${VTK_MODULES_DIR}/vtkPython.cmake)
+            return()
+        endif()
+
+        include(${VTK_MODULES_DIR}/vtkPython.cmake)
+
+        if (NOT "${vtkPython_LIBRARIES}" STREQUAL "${PYTHON_LIBRARIES}")
+            message(SEND_ERROR "Mismatch between VTK's and OpenGeoSys' Python "
+                "libraries: ${vtkPython_LIBRARIES} vs. ${PYTHON_LIBRARIES}. "
+                "This will lead to compilation or linking errors. "
+                "You can fix this error by using the same Python version for "
+                "OpenGeoSys as VTK is built with.")
+        endif()
+    endfunction()
+
+    check_python_version_compatibility()
```

Comment:
Looks good! If it works for you then I am fine. Jenkins will not complain as there are no VTK with Python binding packages used.